src/components/constants/utils.ts

```diff
@@ -0,0 +1,52 @@
+import type { Props } from "astro";
+import type { InferEntrySchema, RenderedContent } from "astro:content";
+import { SITE } from "@/config";
+
+export interface Social {
+  name: string;
+  href: string;
+  linkTitle: string;
+  icon: (_props: Props) => Element;
+}
+
+type CollectionKey = "socials" | "shareLinks";
+
+interface CollectionEntry {
+  id: string;
+  body?: string;
+  collection: CollectionKey;
+  data: InferEntrySchema<CollectionKey>;
+  rendered?: RenderedContent;
+  filePath?: string;
+}
+
+export async function getSocialIcons(
+  list: CollectionEntry[]
+): Promise<Social[]> {
+  const socials: Social[] = [];
+
+  for (const { id, data } of list) {
+    // Import the icon dynamically based on the social id
+    const icon = await import(`./icons/${id}.svg`).catch(() => {
```

Comment:
Hey @satnaing 

Would it be possible to create a function that return all available icons?

Reply:
RN, we can get both `shareLinks` and `socials` using `getSocialIcons()`.
I cannot think where we would need that functionality tho.